### PR TITLE
Autoscaling for 2.0

### PR DESCRIPTION
**Pull request by jdoliner**
This PR is mostly the same code as #5857 with a few additional / different things.

- coefficient parallelism was removed as part of this PR
- the datum task chunking had to be rewritten since the code it was based on had completely changed
- scaling is no longer based on unclaimed tasks but on total tasks, this removes any risk of overscaling due to slow pod spin up or pod crashes.

### CODE REVIEW

**Comment by jdoliner** [comment]
@nadegepepin this has a few implications for docs:
- coefficient was removed, I removed it from the docs as well but some sections may be a bit awkward since they're written like there are two alternatives for parallelism and now there's only 1
- a new field was added to the chunk_spec called `datums_per_worker` I documented it in line but haven't added it to the pipeline spec yet.

File: src/internal/work/work.go [hunk 1]
```diff
@@ -388,3 +391,16 @@ func (w *Worker) subtaskFunc(subtaskKey string, processFunc ProcessFunc) subtask
 		}
 	}
 }
+
+// TaskCount returns how many subtasks are in the queue and how many are claimed.
+func (w *Worker) TaskCount(ctx context.Context) (int64, int64, error) {
```
Comment (brendoncarroll) [inline src/internal/work/work.go:396]: There should probably be names for the 2 int64s, so they can be distinguished by looking at the signature.  Use `_` for the error.

File: src/server/pps/server/monitor.go [hunk 2]
```diff
@@ -325,6 +329,53 @@ func (m *ppsMaster) monitorPipeline(ctx context.Context, pipelineInfo *pps.Pipel
 			}, backoff.NewInfiniteBackOff(),
 				backoff.NotifyCtx(ctx, "monitorPipeline for "+pipeline))
 		})
+		if pipelineInfo.ParallelismSpec != nil && pipelineInfo.ParallelismSpec.Constant > 1 && pipelineInfo.Autoscaling {
+			eg.Go(func() error {
+				pachClient := m.a.env.GetPachClient(ctx)
+				return backoff.RetryNotify(func() error {
```
Comment (brendoncarroll) [inline src/server/pps/server/monitor.go:335]: should this be `RetryUntilCancel` so it can respect the context?

**Review by brendoncarroll — APPROVED** [review]
A couple small suggestions, otherwise LGTM.

File: src/internal/deploy/assets/assets.go [hunk 3]
```diff
@@ -66,7 +66,7 @@ var (
 	}, {
 		APIGroups: []string{""},
 		Verbs:     []string{"get", "list", "watch", "create", "update", "delete"},
-		Resources: []string{"replicationcontrollers", "services"},
+		Resources: []string{"replicationcontrollers", "services", "replicationcontrollers/scale"},
```
Comment (jdoliner) [inline src/internal/deploy/assets/assets.go:69]: @chainlink this is the added resources for autoscaling.

**Pull request merged**